arch/riscv/core/isr.S

```diff
@@ -521,6 +525,45 @@ is_interrupt:
 
 on_irq_stack:
 
+#if defined(CONFIG_NUCLEI_ECLIC) && !defined(CONFIG_RISCV_VECTORED_MODE)
```

Comment:
the only thing about this PR that is triggering me a bit is having an IRQ controller specific `ifdef` (`ECLIC` in this case) in a file that should be not linked to any. 

Now, how difficult would be to have an hook here instead of an `ifdef` to be implemented by the IRQ controller driver to leave this code as generic as possible?

Reply:
done. I've added the kconfig `RISCV_SOC_HAS_CUSTOM_IRQ_HANDLING`. This allows to overwrite the handling of all irqs for a specific SoC. I've moved the new part to soc/riscv/riscv-privileged/common/soc_irq.S. I hope it makes more sense there.